## Data Scientist

In [2]:
import pandas as pd
from sklearn.linear_model import LinearRegression
from sklearn.metrics import r2_score

df = pd.read_csv('air_fryers_clean_brand_year.csv')

feature_cols =[
    'compact_share', 
    'dual_basket_share', 
    'oven_style_share', 
    'rotisserie_share', 
    'window_share'
]

y = df['log_brand_share']

brand_dummies = pd.get_dummies(df['brand'], prefix='brand', drop_first=True, dtype=int)
year_dummies = pd.get_dummies(df['year'].astype(str), prefix='year', drop_first=True, dtype=int)

X = pd.concat([
    df[['avg_price', 'avg_rating'] + feature_cols],
    brand_dummies,
    year_dummies
], axis=1)

model = LinearRegression()
model.fit(X, y)

predicted_log_share = model.predict(X)
r2 = r2_score(y, predicted_log_share)

coef_table = pd.DataFrame({
    'feature': X.columns,
    'coefficient': model.coef_
})

print(f"R-squared: {r2:.4f}\n")
print(coef_table.sort_values(by='coefficient', ascending=False).to_string(index=False))

R-squared: 0.7635

          feature  coefficient
     window_share    12.880298
    compact_share     9.815304
  brand_cuisinart     6.422436
      brand_ninja     5.838705
brand_instant_pot     4.626260
 brand_gowise usa     3.938996
      brand_oster     3.928074
     brand_nuwave     3.544883
     brand_cosori     2.551946
 oven_style_share     1.941774
    brand_ultrean     0.942399
       avg_rating     0.287517
       brand_dash     0.176655
        year_2020     0.119071
        year_2021     0.041900
        year_2023    -0.003307
        avg_price    -0.037668
        year_2022    -0.098860
 rotisserie_share    -5.674054
dual_basket_share    -9.509686


1. **Price Coefficient:** The estimated price coefficient ($\hat{\beta}_{price}$) is -0.0377.
2. **Significance of Negative Sign:** It is negative, which aligns with the law of demand: as price increases, market share decreases. If this were positive, the model would imply consumers prefer paying higher prices, which would completely break the subsequent marginal cost equations.
3. **Valued Features:** The `window_share` (12.88) and `compact_share` (9.82) features are associated with significantly higher demand, followed by `oven_style_share` (1.94). Conversely, dual baskets and rotisseries negatively impact share in this model.
4. **Brand Dummies:** Relative to the dropped brand (Chefman), Cuisinart (6.42), Ninja (5.84), and Instant Pot (4.63) have the highest baseline demand coefficients.
5. **Year Dummies:** The year 2020 has the largest coefficient (0.119), showing a general spike in demand relative to the 2019 baseline, followed by a smaller bump in 2021 (0.042). By 2022 and 2023, baseline demand contracted.
6. **Model Fit:** The model's $R^2$ is 0.7635, meaning it explains about 76% of the variance in log market shares.